src/polyform-license-src/edit/edit.ts

```diff
@@ -0,0 +1,194 @@
+    if (count > 0 && count !== expectedReplacements) {
+        return {
+            content: [{ 
+                type: "text", 
+                text: `Expected ${expectedReplacements} occurrences but found ${count} in ${filePath}. ` + 
+                      `If you want to replace all ${count} occurrences, set expected_replacements to ${count}. ` +
+                      `If you want to replace a specific occurrence, make your search string more unique by adding context.` 
+            }],
+        };
+    }
+    
+    // If exact match not found, try fuzzy search
+    if (count === 0) {
+        // Track fuzzy search time
+        const startTime = performance.now();
+        
+        // Perform fuzzy search
+        const fuzzyResult = recursiveFuzzyIndexOf(content, block.search);
+        const similarity = getSimilarityRatio(block.search, fuzzyResult.value);
+        
+        // Calculate execution time in milliseconds
+        const executionTime = performance.now() - startTime;
+        
+        // Check if the fuzzy match is "close enough"
+        if (similarity >= FUZZY_THRESHOLD) {
+            // Format differences for clearer output
+            const diff = highlightDifferences(block.search, fuzzyResult.value);
+            
+            // Capture the fuzzy search event
+            capture('server_fuzzy_search_performed', {
+                similarity: similarity,
+                execution_time_ms: executionTime,
+                search_length: block.search.length,
+                file_size: content.length,
+                threshold: FUZZY_THRESHOLD,
+                found_text_length: fuzzyResult.value.length
+            });
+            
+            // If we allow fuzzy matches, we would make the replacement here
+            // For now, we'll return a detailed message about the fuzzy match
+            return {
+                content: [{ 
+                    type: "text", 
+                    text: `Exact match not found, but found a similar text with ${Math.round(similarity * 100)}% similarity (found in ${executionTime.toFixed(2)}ms):\n\n` +
+                          `Search string: "${block.search}"\n` +
+                          `Found: "${fuzzyResult.value}"\n\n` +
+                          `Differences:\n${diff}\n\n` +
+                          `To replace this text, use the exact text found in the file.`
+                }],
+            };
+        } else {
+            // If the fuzzy match isn't close enough
+            // Still capture the fuzzy search event even for unsuccessful matches
+            capture('fuzzy_search_performed', {
+                similarity: similarity,
+                execution_time_ms: executionTime,
+                search_length: block.search.length,
+                file_size: content.length,
+                threshold: FUZZY_THRESHOLD,
+                found_text_length: fuzzyResult.value.length,
+                file_path: filePath,
```

Comment:
do we really need to see file path in analytics ? maybe just file extensation?
```suggestion
                file_ext: filePath.split('.').pop(),
```